flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1759,11 +1759,22 @@ class JobManager(
       case None => None
     }
 
-    // remove all job-related BLOBs from local and HA store
-    libraryCacheManager.unregisterJob(jobID)
-    blobServer.cleanupJob(jobID, removeJobFromStateBackend)
+    // remove all job-related BLOBs from local and HA store, only if the job was removed correctly
+    futureOption match {
+      case Some(future) => future.onComplete{
+        case scala.util.Success(_) => {
+          libraryCacheManager.unregisterJob(jobID)
+          blobServer.cleanupJob(jobID, removeJobFromStateBackend)
```

Comment:
Can't we move these this line in the future where we remove the job from the `SubmittedJobGraphStore`?

Reply:
Technically we can, but this changes the return type of the future as `cleanupJob` does indeed return something.